Volume metrics

Henni opened the pull request:
rebases #421
fixes #223, #499 

**Is this a bug fix or adding new feature?**
New Feature

**What is this PR about? / Why do we need it?**
Implementation of `NodeGetVolumeStats ` to report volume metrics:
- `kubelet_volume_stats_available_bytes`
- `kubelet_volume_stats_capacity_bytes`
- `kubelet_volume_stats_inodes`
- `kubelet_volume_stats_inodes_free`
- `kubelet_volume_stats_inodes_used`
- `kubelet_volume_stats_used_bytes`

**What testing is done?** 
Linting is fine.
I wasn't able to fully run the tests yet. I believe this is because the tests only run properly on Linux.

Henni commented:
/check-cla

leakingtapan commented:
@Henni thx for the contribution!
/ok-to-test

gnufied commented on pkg/driver/statter.go:
```diff
@@ -0,0 +1,80 @@
+	inodesFree = int64(statfs.Ffree)
+	inodesUsed = inodes - inodesFree
+	return
+}
```
Is there a way we can reuse code from core k8s to do this rather than copying similar logic here? We can consider moving these utility functions somewhere common where all CSI drivers can use them.

mindw commented:
/remove-lifecycle stale

A requirement achieving production grade?

batas commented:
@Henni could you finish your work? This PR is really nice and would be very helpful in my job :) I saw in other drivers this functionality is successfully implemented.

Henni commented:
This issue is still on my list. I hope to get back to it next week.

Henni commented:
Rebased and removed the statter.go
Instead I am now reusing generic kubernetes code.

cc @matthiaslee @AndyXiangLi
I am ok with applying my changes to any other Pull Request that is currently open (like #677)

Henni commented:
fixed with #677
